src/test/java/org/symphonyoss/symphony/messageml/elements/TextAreaTest.java

```diff
@@ -0,0 +1,216 @@
+    assertEquals(TextArea.class, textArea.getClass());
+
+    verifyTextAreaPresentation((TextArea) textArea, NAME_VALUE, false, true, requiredValue,
+        false, null);
+    verifyTextAreaMarkdown(null);
+  }
+
+  @Test
+  public void testTextAreaPlaceholderAttribute() throws Exception {
+    String input = String.format(
+        "<messageML><form><textarea name=\"%s\" placeholder=\"%s\"></textarea></form></messageML>",
+        NAME_VALUE, PLACEHOLDER_VALUE);
+
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+    Element messageML = context.getMessageML();
+    Element form = messageML.getChildren().get(0);
+    Element textArea = form.getChildren().get(0);
+
+    assertEquals(Form.class, form.getClass());
+    assertEquals(TextArea.class, textArea.getClass());
+
+    verifyTextAreaPresentation((TextArea) textArea, NAME_VALUE, false, false, false, true,
+        PLACEHOLDER_VALUE);
+    verifyTextAreaMarkdown(PLACEHOLDER_VALUE);
+  }
+
+  @Test
+  public void testTextAreaWithoutRequiredFields() throws Exception {
+    String input = "<messageML><form><textarea></textarea></form></messageML>";
+
+    expectedException.expect(InvalidInputException.class);
+    expectedException.expectMessage("The attribute \"name\" is required");
+
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+  }
+
+  @Test
+  public void testTextAreaWithInvalidRequiredAttributeValue() throws Exception {
+    String input = String.format(
+        "<messageML><form><textarea name=\"%s\" required=\"value\"></textarea></form></messageML>",
+        NAME_VALUE);
+
+    expectedException.expect(InvalidInputException.class);
+    expectedException.expectMessage(
+        "Attribute \"required\" of element \"textarea\" can only be one of the following values: "
+            + "[true, false].");
+
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+  }
+
+  @Test
+  public void testTextAreaWithNotAllowedAttribute() throws Exception {
+    String attribute = "anotherAttribute";
+    String input =
+        String.format("<messageML><form><textarea %s=\"value\"></textarea></form></messageML>",
+            attribute);
+
+    expectedException.expect(InvalidInputException.class);
+    expectedException.expectMessage(
+        "Attribute \"" + attribute + "\" is not allowed in \"textarea\"");
+
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+  }
+
+  private void verifyTextAreaPresentation(TextArea textArea, String name,
+      boolean shouldRenderInitialValue, boolean shouldRenderRequired, boolean expectedRequired,
+      boolean shouldRenderPlaceholder, String expectedPlaceholder) {
+
+    assertEquals("Name attribute", name, textArea.getAttribute(NAME_ATTR));
+    assertEquals(
+        getExpectedTextAreaPresentationML(textArea, shouldRenderInitialValue, shouldRenderRequired,
+            shouldRenderPlaceholder), context.getPresentationML());
+
+    if (shouldRenderRequired) {
```

Comment:
While I do agree this check is a good thing to have, I have a concern over the name of this method vs what it actually does due to this check at the end. I know this is just a unit test, but from the method name I don't expect you to check the objects inside the parser other than just the presentationML.

I try to put this comments on other private repositories as well, but with this one the issue is more specific, as by being an opensource code we need to make the most readable possible so others can easily contribute to it.

Reply:
The "concept" of this test scenario is the same as text-field scenario, since the specs are pretty similar - my intention was to keep the same implementation pattern of an already existing test scenario. Anyways, I do agree with your comment. :)

Reply:
Done - removed the object checks.